File: 0001-two-sum/0001-two-sum.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        var map = hashMapOf<Int, Int>()

        for(i in nums.indices) {
            var requiredVal = target - nums[i]
            if(map.containsKey(requiredVal)) {
                return intArrayOf(map[requiredVal]!!, i)
            }

            map[nums[i]] = i
        }

        return intArrayOf(0,0)
    }
}

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dict = {}
        for i in range(0, len(nums)):
            required = target - nums[i]

            if required in dict:
                return [dict[required], i]

            dict[nums[i]] = i